addons/mail/models/mail_message.py

```diff
@@ -1171,10 +1171,16 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
+        messages_per_partner = defaultdict(lambda: self.env['mail.message'])
+        for message in messages:
+            if not self.env.user._is_public():
+                messages_per_partner[self.env.user.partner_id] |= message
+            if message.author_id and not any(user._is_public() for user in message.author_id.user_ids):
```

Comment:
As per change in https://github.com/odoo/odoo/pull/70665, accessing `user_ids` for the purpose of testing `_is_public()` should be done with `active_test=False`